src/tests/ftest/cart/util/cart_utils.py

```diff
@@ -307,12 +308,15 @@ def build_cmd(self, env, host, **kwargs):
 
         index = kwargs.get('index', None)
 
+        daos_test_shared_dir = os.getenv('DAOS_TEST_SHARED_DIR',
+                                         os.getenv('HOME'))
+
         tst_vgd = " valgrind --xml=yes " + \
-                  "--xml-file={}/".format(self.log_path) + \
+                  "--xml-file={}/".format(daos_test_shared_dir) + \
                   r"valgrind.%q\{PMIX_ID\}.memcheck " + \
-                  "--fair-sched=try  --partial-loads-ok=yes " + \
+                  "--fair-sched=try --partial-loads-ok=yes " + \
                   "--leak-check=yes --gen-suppressions=all " + \
-                  "--suppressions=../etc/memcheck-cart.supp " + \
+                  "--suppressions=" + self.supp_file + " " + \
```

Comment:
For NLT the code we use is at https://github.com/daos-stack/daos/blob/master/utils/node_local_test.py#L888-L912

I think partial-loads-ok=yes was needed on older CentOS 7 systems but we don't appear to use it any more, NLT uses --leak-check=full, --show-leak-kinds=all which appears to be stricter than what's here.  fair-sched=try means to attempt fair scheduling or back off, fair-sched=yes means to abort if it can't be used which is preferred.

The other change is the --error-exitcode=42 which we use to force NLT to fail (in most cases) rather than relying purely on the parsing of xml files in Valgrind.

Reply:
> The other change is the --error-exitcode=42 which we use to force NLT to fail (in most cases) rather than relying purely on the parsing of xml files in Valgrind.

`crt_launch` seems to be _"eating"_ the `42` exit status.  The below example runs a simple 2-line C program that leaks memory.  Without `crt_launch`, `orterun` returns `42`:
```
$ sh -x run.sh
+ gcc uninitialized-memory.c
+ orterun valgrind --quiet --leak-check=full --error-exitcode=42 ./a.out
This program intentionally leaks memory.
This program intentionally leaks memory.
==1480== 10 bytes in 1 blocks are definitely lost in loss record 1 of 1
==1480==    at 0x4C29F73: malloc (vg_replace_malloc.c:309)
==1480==    by 0x400598: main (in /home/emallovx/src/valgrind-leaks-sandbox/a.out)
==1480==
==1479== 10 bytes in 1 blocks are definitely lost in loss record 1 of 1
==1479==    at 0x4C29F73: malloc (vg_replace_malloc.c:309)
==1479==    by 0x400598: main (in /home/emallovx/src/valgrind-leaks-sandbox/a.out)
==1479==
...
--------------------------------------------------------------------------
orterun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[52472,1],1]
  Exit code:    42
--------------------------------------------------------------------------
+ echo '? = 42'
? = 42
```

*With* `crt_launch`, `orterun` returns `0`:
```
+ orterun valgrind --quiet --leak-check=full --error-exitcode=42 crt_launch -c -e ./a.out
...
This program intentionally leaks memory.
This program intentionally leaks memory.
+ echo '? = 0'
? = 0
$
```

Reply:
`--trace-children=yes` option to Valgrind seems to resolve this problem.

Reply:
[CART-977](https://daosio.atlassian.net/browse/CART-977) addresses return code handling.